src/main/java/frc/robot/subsystem/DriveSubsystem.java

```diff
@@ -216,17 +218,29 @@ public void driveCartesian(
                 turnRequest * maxAngularSpeed,
                 fieldOrientedDriveAngle);
 
+        driveRobotRelativeSpeeds(speeds);
+    }
+
+    public void driveRobotRelativeSpeeds(ChassisSpeeds speeds) {
         driveSwerveStates(kinematics.toSwerveModuleStates(speeds));
+
+        robotRelativeSpeeds = speeds;
+    }
+
+    public ChassisSpeeds getLastRobotRelativeSpeeds() {
+        return robotRelativeSpeeds;
     }
```

Comment:
This isn't right. The `Supplier<ChassisSpeeds>` required by the `FollowPathHolonomic` command should return the **actual** current chassis speeds of the robot.

Think, if the `FollowPathHolonomic` implementation just wanted to know whatever it commanded on the last iteration, it could just keep track of that itself. The point of adding the external Supplier is so you can add the logic to get the actual speeds.

We have a `SwerveDriveKinematics` instance. You should be using [toChassisSpeeds](https://github.wpilib.org/allwpilib/docs/release/java/edu/wpi/first/math/kinematics/SwerveDriveKinematics.html#toChassisSpeeds(edu.wpi.first.math.kinematics.SwerveModuleState...)). Add a `getState()` method to the `SwerveModule` class that returns a `SwerveModuleState` (similar to the `getPosition()` method that's already there), pass those states into `toChassisSpeeds()`, and return the results.

You should also update the name of this method and remove the "last", since it's returning the current state of the robot, not the last commanded state. Call it `getCurrentRobotRelativeSpeeds()` or just `getRobotRelativeSpeeds()`.